students/Gabe_Lamberth/lesson07/build_mailroom.py

```diff
@@ -2,101 +2,184 @@
+    """
+        Add a new donor to the donor database
+
+    """
+    # Create variables to provide random 3 digit id for Donor primary key
+    x = random.randrange(1, 9)
+    y = random.randrange(1, 9)
+    z = random.randrange(1, 9)
+    value = str(x) + str(y) + str(z)
+
+    pri_ID = int(value) # will be inserted as first value of new record for Donor and Donation
+    donor = [] # Used to hold user input
+
+    first_name = 0
+    last_name = 1
+    state = 2
+    job_type = 3
+
+    logging.basicConfig(level=logging.INFO)
+    logger = logging.getLogger(__name__)
+
+    while True:
+        record = input("""
+                       Enter donor data in the following format:
+                       'First Name, Last Name, State, Job Type'
+                       ex: Samuel, Jackson, NY, Actor
+                       Type quit - to exit to main menu.
+                       Enter input here >>  """)
+        if 'quit' in record.lower():
+            break
+        else:
+            for data in record.split(','):
+                donor.append(data)
+
+            logger.info('Connecting to database....')
+            database = SqliteDatabase('donor.db')
+            try:
+                database.connect()
+                database.execute_sql('PRAGMA foreign_keys = ON;')
+
+                logger.info('Add and display a new donor name...')
+                new_person = Donor.create(
+                    donor_ID=pri_ID,
+                    first_name=donor[first_name],
+                    last_name=donor[last_name],
+                    state=donor[state],
+                    job_type=donor[job_type],
+                )
+                new_person.save()
+                logger.info('Show new donor name')
+                a_person = Donor.get(Donor.donor_ID == pri_ID)
+                logger.info(f'We just created {a_person.first_name}, {a_person.last_name}')
+
+            except Exception as e:
+                logger.info(e)
+
+            finally:
+                database.close()
+
+def add_donation(pri_key):
+    # Create variables to provide random 3 digit id for Donor primary key
+    x = random.randrange(1, 9)
+    y = random.randrange(1, 9)
+    z = random.randrange(1, 9)
+    value = str(x) + str(y) + str(z)
+
+    pri_id = int(value)  # will be inserted as first value of new record for Donor and Donation
+
+    now = str(date.today())
+
+    logging.basicConfig(level=logging.INFO)
+    logger = logging.getLogger(__name__)
+
+    while True:
+        amount = input("""
+                           Enter donation amount and any notes in the following format:
+                           '0.00, ex: 200.50                          
+                           Enter input here >>  """)
+        notes = input("Enter any notes for donation >> ")
+
+        try:
+            logger.info('Adding donation...')
+            new_person = Donation.create(
+                money_ID=pri_id,
+                donor_ID=pri_key,
```

Comment:
why not name these vars `money_id` and `donor_id` instead of `pri_key` in the function call `def add_donation(pri_key)`.  The var name `pri_key` doesn't tell us what its a primary key for.